lib/ansible/modules/storage/netapp/sf_volume_manager.py

```diff
@@ -0,0 +1,406 @@
+#!/usr/bin/python
+# (c) 2016, NetApp, Inc
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+#!/usr/bin/python
+
+DOCUMENTATION = '''
+
+module: sf_volume_manager
+
+short_description: Manage SolidFire volumes
+extends_documentation_fragment:
+    - netapp.solidfire
+version_added: '2.3'
+author: Sumit Kumar (sumit4@netapp.com)
+description:
+- Create, destroy, or update volumes on SolidFire
+    - auth_basic
+
+options:
+
+    state:
+        required: true
+        description:
+        - Whether the specified volume should exist or not.
+        choices: ['present', 'absent']
+
+    name:
+        required: true
+        description:
+        - The name of the volume to manage
+
+    account_id:
+        required: true
+        type: int
+        description:
+        - account_id for the owner of this volume
+
+    new_account_id:
+        required: false
+        type: int
+        description:
+        - new account_id for the volume
+
+    enable512e:
+        required: false
+        note: required when state == 'present'
+        type: bool
+        description:
+        - Should the volume provides 512-byte sector emulation?
+
+    qos:
+        required: false
+        type: QoS object
+        description: Initial quality of service settings for this volume.
+
+    attributes:
+        required: false
+        type: dict
+        description: List of Name/Value pairs in JSON object format.
+
+    volume_id:
```

Comment:
Is this necessary given account_id is a required parameter?  My guidance is to eliminate this parameter and clearly comment why volume names can be used.  
Along the same lines, it's always s fine line between doing what's easiest in the api and whats easier for the playbook developer.   It's not a must-fix for this PR, but I would like to understand the impact to this module (probably all SF modules) and the associated playbooks of allowing account names to be used instead of account id's.  Are the playbooks easier to read or develop?  Or does this module nicety amount to needless lines of code in the module?  Getting a module out in to the community now is more important than making this decision.